File: domain/src/testFixtures/java/com/study/platform/support/fake/FakeTeamScheduleRepository.java
package com.study.platform.support.fake;

import com.study.platform.domain.team.model.TeamSchedule;
import com.study.platform.domain.team.model.TeamScheduleRepository;
import org.springframework.test.util.ReflectionTestUtils;

import java.time.LocalDateTime;
import java.util.Comparator;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Optional;
import java.util.UUID;

public class FakeTeamScheduleRepository implements TeamScheduleRepository {

    private final Map<UUID, TeamSchedule> store = new HashMap<>();

    @Override
    public TeamSchedule save(TeamSchedule schedule) {
        if (schedule.getId() == null) {
            ReflectionTestUtils.setField(schedule, "id", UUID.randomUUID());
        }
        store.put(schedule.getId(), schedule);
        return schedule;
    }

    @Override
    public Optional<TeamSchedule> findById(UUID id) {
        return Optional.ofNullable(store.get(id));
    }

    @Override
    public void delete(TeamSchedule schedule) {
        store.remove(schedule.getId());
    }

    @Override
    public List<TeamSchedule> findAllByTeamIdOrderByScheduledAtAsc(UUID teamId) {
        return store.values().stream()
                .filter(s -> s.getTeam().getId().equals(teamId))
                .sorted(Comparator.comparing(TeamSchedule::getScheduledAt))
                .toList();
    }

    @Override
    public List<TeamSchedule> findAllByScheduledAtBetweenWithTeam(LocalDateTime start, LocalDateTime end) {
        return store.values().stream()
                .filter(s -> !s.getScheduledAt().isBefore(start) && !s.getScheduledAt().isAfter(end))
                .toList();
    }
}

File: domain/src/testFixtures/kotlin/com/study/platform/support/fake/FakeTeamScheduleRepository.kt
package com.study.platform.support.fake

import com.study.platform.domain.team.model.TeamSchedule
import com.study.platform.domain.team.model.TeamScheduleRepository
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Slice
import org.springframework.data.domain.SliceImpl
import java.time.LocalDateTime
import java.util.UUID

class FakeTeamScheduleRepository : AbstractFakeUuidRepository<TeamSchedule>(), TeamScheduleRepository {

    override fun idOf(entity: TeamSchedule): UUID? = entity.id
    override fun hasTimestamps() = true

    override fun save(schedule: TeamSchedule): TeamSchedule = saveEntity(schedule)

    override fun delete(schedule: TeamSchedule) = deleteEntity(schedule)

    override fun deleteAllByTeamId(teamId: UUID) {
        store.values.filter { it.team?.id == teamId }.mapNotNull { it.id }.forEach { store.remove(it) }
    }

    fun findAllByTeamId(teamId: UUID): List<TeamSchedule> =
        store.values
            .filter { s -> s.team?.id == teamId }
            .sortedBy { it.scheduledAt }

    override fun findAllByScheduledAtBetweenWithTeam(start: LocalDateTime, end: LocalDateTime, pageable: Pageable): Slice<TeamSchedule> {
        val matched = store.values
            .filter { s -> s.scheduledAt?.let { !it.isBefore(start) && !it.isAfter(end) } == true }
            .sortedBy { it.id.toString() }
        val content = matched.drop(pageable.offset.toInt()).take(pageable.pageSize)
        val hasNext = pageable.offset + content.size < matched.size
        return SliceImpl(content, pageable, hasNext)
    }
}
